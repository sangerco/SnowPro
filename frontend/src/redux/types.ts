export interface UserData {
    username: string;
    firstName: string;
    lastName: string;
    email: string;
    avatar: string;
    bio: string;
    videos: string[];
    photos: string[];
    favMountains: string[];
  }


export const FETCH_USER_DATA_REQUEST = 'FETCH_USER_DATA_REQUEST';
export const FETCH_USER_DATA_SUCCESS = 'FETCH_USER_DATA_SUCCESS';
export const FETCH_USER_DATA_FAILURE = 'FETCH_USER_DATA_FAILURE';


export interface FetchUserDataRequestAction {
    type: typeof FETCH_USER_DATA_REQUEST
}

export interface FetchUserDataSuccessAction {
    type: typeof FETCH_USER_DATA_SUCCESS;
    payload: UserData;
}

export interface FetchUserDataFailureAction {
    type: typeof FETCH_USER_DATA_FAILURE;
    payload: string;
}

export type UserActionTypes = 
    FetchUserDataRequestAction
    | FetchUserDataSuccessAction
    | FetchUserDataFailureAction;