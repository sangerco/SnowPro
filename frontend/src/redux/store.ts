import { Action, compose } from "redux";
import { ThunkAction, configureStore } from "@reduxjs/toolkit";
import { logger } from "redux-logger";
import rootReducer from "./reducers/rootReducer";

const composeEnhancers =
  (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(logger, composeEnhancers),
});
