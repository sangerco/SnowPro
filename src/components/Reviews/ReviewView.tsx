import React from "react";
import { Container, Header, Divider, Rating } from "semantic-ui-react";
import { Link } from "react-router-dom";

const ReviewView: React.FC = ({ review }) => {
  const formatDate = (date: Date) => {
    const newDate = new Date(date);
    const options: Intl.DateTimeFormatOptions = {
      year: "numeric",
      month: "long",
      day: "numeric",
    };
    return newDate.toLocaleDateString("en-US", options);
  };

  return (
    <Container as={Link} to={`/ski-areas/reviews/${review.id}`}>
      <Header as="h2">{review.header}</Header>
      <Rating icon="star" defaultRating={review.stars} maxRating={5} disabled />
      <Header as="h6" textAlign="right">
        {review.date}
      </Header>
    </Container>
  );
};

export default ReviewView;
