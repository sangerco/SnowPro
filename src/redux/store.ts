import { configureStore, combineReducers } from "@reduxjs/toolkit";
import userReducer from "./slices/userSlice";
import authReducer from "./slices/authSlice";
import messageReducer from "./slices/messageSlice";
import messageReplyReducer from "./slices/messageReplySlice";
import reviewReducer from "./slices/reviewSlice";
import reviewReplyReducer from "./slices/reviewReplySlice";
import tagReducer from "./slices/tagSlice";
import storage from "redux-persist/lib/storage";
import { persistReducer, persistStore } from "redux-persist";

const persistConfig = {
  key: "root",
  storage,
  blacklist: ["messages", "messageReplies", "reviews", "reviewReplies"],
};

const rootReducer = combineReducers({
  users: userReducer,
  auth: authReducer,
  messages: messageReducer,
  messageReplies: messageReplyReducer,
  reviews: reviewReducer,
  reviewReplies: reviewReplyReducer,
  tags: tagReducer,
});

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
  reducer: persistedReducer,
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export const persistor = persistStore(store);
