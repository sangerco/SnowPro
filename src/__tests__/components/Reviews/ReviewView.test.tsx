/* eslint-disable testing-library/no-node-access */
import React from "react";
import ReviewView from "../../../components/Reviews/ReviewView";
import { render, fireEvent, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "../../../redux/store";
import { BrowserRouter } from "react-router-dom";

const testReview = {
  id: "review1",
  userId: "11",
  username: "testuser",
  skiAreaSlug: "skiArea1",
  skiAreaName: "Test Ski Area",
  header: "Test Review",
  body: "Test Review Body",
  stars: 3,
  photos: [],
  createdAt: new Date(),
  replies: [],
};

describe("test Review View", () => {
  it("should render without crashing", () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <ReviewView review={testReview} />
        </BrowserRouter>
      </Provider>
    );
  });
  it("should match snapshot", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <ReviewView review={testReview} />
        </BrowserRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("should contain the correct data", () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <ReviewView review={testReview} />
        </BrowserRouter>
      </Provider>
    );

    const header = screen.getByTestId("review-header");
    // @ts-ignore
    expect(header.textContent).toEqual("Test Review");
  });
});
