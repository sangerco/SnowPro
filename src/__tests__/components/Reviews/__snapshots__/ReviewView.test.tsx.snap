// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Nav Bar should match snapshot 1`] = `
<DocumentFragment>
  <a
    class="ui container"
    href="/ski-areas/reviews/review1"
  >
    <h2
      class="ui header"
    >
      Test Review
    </h2>
    <div
      class="ui star disabled rating"
      role="radiogroup"
      tabindex="0"
    >
      <i
        aria-checked="false"
        aria-posinset="1"
        aria-setsize="5"
        class="active icon"
        role="radio"
        tabindex="-1"
      />
      <i
        aria-checked="false"
        aria-posinset="2"
        aria-setsize="5"
        class="active icon"
        role="radio"
        tabindex="-1"
      />
      <i
        aria-checked="true"
        aria-posinset="3"
        aria-setsize="5"
        class="active icon"
        role="radio"
        tabindex="-1"
      />
      <i
        aria-checked="false"
        aria-posinset="4"
        aria-setsize="5"
        class="icon"
        role="radio"
        tabindex="-1"
      />
      <i
        aria-checked="false"
        aria-posinset="5"
        aria-setsize="5"
        class="icon"
        role="radio"
        tabindex="-1"
      />
    </div>
    <h6
      class="ui right aligned header"
    >
      October 1, 2023
    </h6>
  </a>
</DocumentFragment>
`;
